plugins/cisco_firepower_management_center/icon_cisco_firepower_management_center/actions/block_url_policy/action.py

```diff
@@ -16,62 +15,105 @@ def __init__(self):
-        if "paging" in policy:
-            if policy.get("paging", {}).get("count") == 0:
-                policy = acp.post()
-        self.logger.info(policy)
-        return policy
-
-    @staticmethod
-    def _make_url_object(fmc: fmcapi.FMC, name: str, fqdn: str) -> str:
-        url = fmcapi.URLs(fmc=fmc)
-        url.name = name
-        url.url = fqdn
-        url.post()
-        url.get()
-        return url.id
-
-    @staticmethod
-    def _make_rule(fmc: fmcapi.FMC, policy: dict, rule_name: str, urls: dict) -> str:
-        acr = fmcapi.AccessRules(fmc=fmc)
-        acr.name = rule_name
-        acr.urls = urls
-        acr.acp_id = policy.get("id")
-        acr.URL = policy.get("rules", {}).get("links", {}).get("self")
-        acr.action = "BLOCK"
-        acr.enabled = True
-        acr.post()
-        return acr.id
+    def get_policy_object(self, policy_name: str) -> Union[dict, None]:
+        """
+        A method to get the individual policy object if it is found, otherwise
+        return None
+
+        :param policy_name: Name of the policy
+
+        :return: Policy object if found, else None
+        """
+
+        policies = self.connection.cisco_firepower_api.get_policies()
+        # If get AND found, return whole individual policy object
+        for policy in policies.get("items", []):
+            if policy.get("name") == policy_name:
+                return self.connection.cisco_firepower_api.get_policies(policy.get("id", ""))
+        return None
+
+    def make_policy(self, policy_name: str) -> dict:
+        """
+        A method to create a new policy object with the given name if not found,
+        else return the policy object matching the input name
+
+        :param policy_name: Name of the policy
+
+        :return: Policy object
+        """
+
+        policy_obj = self.get_policy_object(policy_name)
+
+        if policy_obj is None:
+            self.logger.info(f"No policy found, creating new policy: {policy_name}")
+            return self.connection.cisco_firepower_api.post_policy(
+                {"name": policy_name, "type": "AccessPolicy", "defaultAction": {"action": "BLOCK"}}
+            )
+        else:
+            self.logger.info(f"Found policy returned for: {policy_name}")
+            return policy_obj
+
+    def make_url_object(self, name: str, url: str) -> str:
+        """
+        A method to post the input URLs to add them to the db,
+        then get the URLs and return the corresponding, newly created ID for each.
+
+        :param name: URL Name, e.g. google
+        :param url: URL, e.g. https://www.google.com
+
+        :return: Unique ID associated with URL
+        """
+        payload = {"name": name, "url": url}
+        self.connection.cisco_firepower_api.post_urls(payload)
+
+        response = self.connection.cisco_firepower_api.get_urls()
+        url_list = response.get("items")
+
+        for item in url_list:
+            if item.get("name") == name:
```

Comment:
this would be better as

`item.get("name", "")`

as if there is no value for name it will try and compare `None` to STR and error out